logstash-core/src/main/java/org/logstash/launchers/JvmOptionsParser.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    /**
+     * The main entry point. The exit code is 0 if the JVM options were successfully parsed, otherwise the exit code is 1. If an improperly
+     * formatted line is discovered, the line is output to standard error.
+     *
+     * @param args the args to the program which should consist of a single option, the path to LOGSTASH_HOME
+     */
+    public static void main(final String[] args) throws InterruptedException, IOException {
+        if (args.length != 1) {
+            throw new IllegalArgumentException(
+                    "Expected one arguments specifying path to LOGSTASH_HOME, but was " + Arrays.toString(args)
+            );
+        }
+
+        final JvmOptionsParser parser = new JvmOptionsParser(args[0]);
+        try {
+            parser.parse();
+        } catch (JvmOptionsFileParserException pex) {
+            System.err.printf(Locale.ROOT,
+                    "encountered [%d] error%s parsing [%s]",
+                    pex.invalidLines().size(),
+                    pex.invalidLines().size() == 1 ? "" : "s",
+                    pex.jvmOptionsFile());
+            int errorCounter = 0;
+            for (final Map.Entry<Integer, String> entry : pex.invalidLines().entrySet()) {
+                errorCounter++;
+                System.err.printf(Locale.ROOT,
+                        "[%d]: encountered improperly formatted JVM option in [%s] on line number [%d]: [%s]",
+                        errorCounter,
+                        pex.jvmOptionsFile(),
+                        entry.getKey(),
+                        entry.getValue());
+            }
+        } catch (IOException ex) {
+            System.err.println("Error accessing jvm.options file");
+            System.exit(1);
+        }
+    }
+
+    private void parse() throws IOException, JvmOptionsFileParserException {
+        final String jvmOpts = System.getenv("LS_JVM_OPTS");
+        // Set the initial JVM options from the jvm.options file. Look in
+        // /etc/logstash first, and break if that file is found readable there.
+        Optional<Path> jvmOptions;
+        if (jvmOpts == null || jvmOpts.isEmpty()) {
+            jvmOptions = Arrays.stream(new Path[] { Paths.get("/etc/logstash/jvm.options"),
+                                                    Paths.get(logstashHome, "/config/jvm.options") })
+                    .filter(p -> p.toFile().canRead()).findFirst();
+        } else {
+            jvmOptions = Optional.of(Paths.get(jvmOpts));
+        }
+        if (jvmOptions.isPresent()) {
+            final List<String> jvmOptionsContent = parseJvmOptions(jvmOptions.get());
+            final String lsJavaOpts = System.getenv("LS_JAVA_OPTS");
+            if (lsJavaOpts != null && !lsJavaOpts.isEmpty()) {
+                jvmOptionsContent.add(lsJavaOpts);
+            }
+            System.out.println(String.join(" ", jvmOptionsContent));
+        } else {
+            System.err.println("warning: no jvm.options file found");
```

Comment:
nit: this method is called `parse`, but its implementation is almost exclusively about _finding_ the file to parse

Reply:
I split this in 2 methods:
- `lookupJvmOptionsFile`: that lookups for the `jvm.options` file to parse, returning an Optional of Path
- 'parse(Path)': which parses the given jvm.option file